Mobile-friendly CSS

himynameisdave opened the pull request:
Adds mobile-friendly/responsive CSS for all views. All CSS is now in static CSS files (so they can be cached by the browser). Also breaks some of the templates off into smaller bits, for easier reuse. 

Ready for review, open to any feedback here (as I got a bit carried away with some of these).

Resolves #29

poundifdef commented:
Thank you for this! I pulled down the code and looked at it locally, and I really like the improvements for mobile. That said, I have a few comments/suggestions:

1. The printer-specific styles in your PR are different than the original. Was that intentional? Perhaps you have some guidance, but my overall goal was for the recipe cards to be more dense than the original, such that they could (if possible) fit on the front and back of an index card.

2. The code, as written, does have a lot of if-statements regarding which templates, and which "fragments", are loaded on the home page versus the recipe page. Could this be simplified? Even if there is repeated code (for example, one could do a`header.html` and `recipe-header.html`), I think that would make it much easier to follow, rather than needing to keep a mental model of different possible states and conditionals when looking at code.

Really appreciate this!

himynameisdave commented:
Thanks for the feedback!

> _The printer-specific styles in your PR are different than the original. Was that intentional? Perhaps you have some guidance, but my overall goal was for the recipe cards to be more dense than the original, such that they could (if possible) fit on the front and back of an index card._

So the only thing I removed was the commented out `font-family` declaration [here](https://github.com/poundifdef/plainoldrecipe/blob/master/templates/recipe.html#L17-L19), however I think that it is inheriting some of the global/mobile CSS from the page, which is breaking the number of columns (as well as some other stuff). I've corrected this, and the `print` styles are now basically identical to how they are currently:

![print-styles](https://user-images.githubusercontent.com/4298089/90457529-b001df00-e0b0-11ea-87a9-798114ecabfe.png)

> _The code, as written, does have a lot of if-statements regarding which templates, and which "fragments", are loaded on the home page versus the recipe page. Could this be simplified? Even if there is repeated code (for example, one could do a `header.html` and `recipe-header.html`), I think that would make it much easier to follow, rather than needing to keep a mental model of different possible states and conditionals when looking at code._

Yep, totally agree. I think I went a bit overboard with [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) 😄. As you said, a lot of this was around the differences between the recipe header and the main header, so I've just split them in two for now. Let me know if you think there's anything other places where I should remove conditionals/simplify, or if there are any other changes I should make in order to get this PR merged. Thanks! 💪
